src/ColinHDev/CPlot/provider/DataProvider.php

```diff
@@ -164,6 +175,10 @@ private function initializeDatabase() : Generator {
         yield from Await::all($generators);
         yield from $this->database->asyncGeneric(self::INIT_ASTERISK_PLAYER, ["lastJoin" => date("d.m.Y H:i:s")]);
         $this->isInitialized = true;
+
+        if(ResourceManager::getInstance()->getConfig()->getNested("database.import", false) === true) {
+            CPlot::getInstance()->getScheduler()->scheduleTask(new ClosureTask(\Closure::fromCallable(fn() => Await::f2c(\Closure::fromCallable([$this, "importData"]))))); // 1 tick delay to ensure worlds are all loaded
+        }
```

Comment:
Hmm... So you propose the conversion should take place on server start-up?
My original idea was to do this when the world is loaded.

Reply:
It can be done then, but I figured organizationally it makes more sense to keep all the database related code in the same class. It can totally be changed if you'd like.

Reply:
Yes, the method should stay in the `DataProvider` class.
We can't import the plot data on server start-up, since we only import the world data once the world gets loaded. The `plots` table's `worldName` column references an entry in the `worlds` table through a foreign key. 
If we would move the plot conversion to the MyPlotConversionListener, we would have to make the `importData()` method public, which is not ideal, since that method is only for internal use and should not be accessible to others. But on the other hand, the `DataProvider` is not meant to be used by others as an API. (There is a reason the  `CPlotAPI` class was created.)

Reply:
This works as intended anyways. Converted worlds are all loaded and registered in the first tick, then the callback is initiated, so this function only gets run after all startup worlds are loaded.

Reply:
But there is no guarantee that the `MyPlotConversionListener` has finished already, or am I wrong here?

Reply:
It's guaranteed because all worlds are loaded within the first tick